src/http/controllers/submissions_controller.rs

```diff
@@ -0,0 +1,54 @@
+use actix_web::{get, HttpResponse, post, Responder, Result, web};
+use charybdis::operations::{Find, Insert};
+use charybdis::types::Uuid;
+use serde_json::json;
+use validator::Validate;
+
+use crate::AppState;
+use crate::http::FuckThatError;
+use crate::http::requests::submission::{SubmissionDTO, SubmissionResponse};
+use crate::models::leaderboard::Leaderboard;
+use crate::models::submission::Submission;
+
+
+#[get("/submissions/{id}")]
+async fn get_submission(
+    data: web::Data<AppState>,
+    id: web::Path<Uuid>,
+) -> Result<impl Responder, FuckThatError> {
+    let submission = Submission { id: id.into_inner(), ..Default::default() };
+    let submission = submission
+        .find_by_primary_key(&data.database)
+        .await;

```

Comment:
I noticed here that you don't pass `played_at` so this is not gonna work unless `played_at` is default value.

In general this is good way to do `find_by_primary_key`, especially when we have complex pk structure. Alternatively, in case of Submission as we have only 2 keys, so we could use automatically generated `let submission = Submission::find_by_id_and_played_at(session, id, played_at)`.